lib/src/firebase_image.dart

```diff
@@ -61,7 +61,7 @@ class FirebaseImage extends ImageProvider<FirebaseImage> {
 
   static String _getImagePath(String location) {
     final uri = Uri.parse(location);
-    return uri.path;
+    return uri.path.replaceAll('%20', ' ');
```

Comment:
In testing, I've found an issue where if the image has "%20" in the filename, it will replace that with a space as well. Any ideas?

Reply:
Just an idea. It maybe not the best, but replace "%20" with something completely unique value like "|*-*|" or anything (which you think shouldn't be in filename or pathname) prior to parse which converts spaces to "%20". And then lastly replace back all the occurence of your uniquely added string with "%20".